From 5c9b60abfd9d79ce10974d1b97ba861fd2bd020a Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 3 Jul 2024 19:12:30 +0200
Subject: [PATCH 651/954] feat(stm32mp2): display package type for STM32MP23

The package type is read from OTP fuse. Display it with other SoC
information at boot.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git fcaf2ffab58866d2765397bd5c30ffb6cc079d6c feat(stm32mp2): display package type for STM32MP23]
Change-Id: I855bd1ac8d96ca88a7e4ee5ac8b6f65697214cc9
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/391283
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h     |  6 ++++++
 plat/st/stm32mp2/stm32mp2_private.c | 14 ++++++++++++++
 2 files changed, 20 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index 7dfc9e083c..404d97cdc5 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -124,6 +124,12 @@
 #define STM32MP21_PKG_AO_VFBGA225	U(4)
 #define STM32MP21_PKG_AM_TFBGA289	U(5)
 #endif /* STM32MP21 */
+#if STM32MP23
+#define STM32MP23_PKG_CUSTOM		U(0)
+#define STM32MP23_PKG_AL_VFBGA361	U(1)
+#define STM32MP23_PKG_AK_VFBGA424	U(3)
+#define STM32MP23_PKG_AJ_TFBGA361	U(7)
+#endif /* STM32MP23 */
 #if STM32MP25
 #define STM32MP25_PKG_CUSTOM		U(0)
 #define STM32MP25_PKG_AL_VFBGA361	U(1)
diff --git a/plat/st/stm32mp2/stm32mp2_private.c b/plat/st/stm32mp2/stm32mp2_private.c
index 2f8fe2c4fe..1fe89f2e4c 100644
--- a/plat/st/stm32mp2/stm32mp2_private.c
+++ b/plat/st/stm32mp2/stm32mp2_private.c
@@ -399,6 +399,20 @@ void stm32mp_get_soc_name(char name[STM32_SOC_NAME_SIZE])
 		pkg = "AM";
 		break;
 #endif /* STM32MP21 */
+#if STM32MP23
+	case STM32MP23_PKG_CUSTOM:
+		pkg = "XX";
+		break;
+	case STM32MP23_PKG_AL_VFBGA361:
+		pkg = "AL";
+		break;
+	case STM32MP23_PKG_AK_VFBGA424:
+		pkg = "AK";
+		break;
+	case STM32MP23_PKG_AJ_TFBGA361:
+		pkg = "AJ";
+		break;
+#endif /* STM32MP23 */
 #if STM32MP25
 	case STM32MP25_PKG_CUSTOM:
 		pkg = "XX";
-- 
2.53.0

